pkg/controller/controller.go

```diff
@@ -28,6 +28,7 @@ type AppGwIngressController struct {
 
 	eventQueue *EventQueue
 
+	configCache *[]byte
```

Comment:
Did you test it by running against a cluster or a mock kubernetes client ?

Reply:
Yes, Sir!
Tested it w/ my AKS. Works like a charm.
Here's an example:
```
I0523 16:35:04.657645   27362 controller.go:118] BEGIN ApplicationGateway deployment
I0523 16:35:04.835279   27362 controller.go:131] deployment took 177.596ms
I0523 16:35:04.835310   27362 controller.go:138] END ApplicationGateway deployment
I0523 16:35:04.835326   27362 eventqueue.go:134] Processing event done, updating lastEventTimestamp
I0523 16:35:04.835368   27362 eventqueue.go:127] Processing event begin, time since event generation: 5.3067173s
I0523 16:35:04.835403   27362 controller.go:53] controller.Process called with type k8scontext.Event
I0523 16:35:04.902868   27362 health_probes.go:25] [health-probes] Configuring health probes for ingress: 'websocket-repeater'
I0523 16:35:04.902901   27362 health_probes.go:33] [health-probes] Working on ingress rule #1: host='.............'
I0523 16:35:04.902915   27362 health_probes.go:41] [health-probes] Working on path #1: ''
I0523 16:35:04.902932   27362 health_probes.go:25] [health-probes] Configuring health probes for ingress: 'ws-ingress'
I0523 16:35:04.902962   27362 health_probes.go:33] [health-probes] Working on ingress rule #1: host='................'
I0523 16:35:04.902979   27362 health_probes.go:41] [health-probes] Working on path #1: ''
I0523 16:35:04.902994   27362 health_probes.go:50] [health-probes] Adding default probe: 'defaultprobe'
I0523 16:35:04.903035   27362 health_probes.go:58] [health-probes] Found k8s probe for backend: 'websocket-repeater'
I0523 16:35:04.903068   27362 health_probes.go:58] [health-probes] Found k8s probe for backend: 'ws-service'
I0523 16:35:04.903085   27362 health_probes.go:67] [health-probes] Will create 3 App Gateway probes.
I0523 16:35:04.903118   27362 backendhttpsettings.go:141] Created a new HTTP setting w/ name: bp-default-websocket-repeater-80-8888-websocket-repeater
I0523 16:35:04.903138   27362 backendhttpsettings.go:141] Created a new HTTP setting w/ name: bp-default-ws-service-80-8888-ws-ingress
I0523 16:35:04.903305   27362 redirects.go:25] Created redirection configuration; not attached to a routing rule yet. Configuration: ............
I0523 16:35:04.903446   27362 requestroutingrules.go:258] Attaching redirection config ..........
I0523 16:35:04.904113   27362 controller.go:114] Config has NOT changed! No need to connect to ARM.
I0523 16:35:04.904129   27362 eventqueue.go:134] Processing event done, updating lastEventTimestamp
```

Note: `I0523 16:35:04.904113   27362 controller.go:114] Config has NOT changed! No need to connect to ARM.`